hazelcast/src/main/java/com/hazelcast/map/impl/record/SimpleRecord.java

```diff
@@ -70,6 +70,11 @@ public boolean casCachedValue(Object expectedValue, Object newValue) {
         return true;
     }
 
+    @Override
+    public int getHits() {
```

Comment:
Shouldn't we change `Record#getHits` to return `0` as the default value?

Reply:
Yes, that makes sense and is most intuitive. But it will be the exception from the `Record` class logic.
Let's @ahmetmircik comment on this, as a `Record` class author.

Reply:
From `UNSET` javadoc:
```
     * Represents an unset value. This is the default
     * value of ttl, max-idle or anything unavailable.
```
`hits` is a different kind of field than `ttl`, `max-idle` etc. It's just a counter AFAIK, so in the initial state it's natural to be equal to `0`. I think that `hits` is never `unavailable`

Reply:
Good point!

Reply:
UNSET is used at places where no such stats is expected, by returning zero we loss this info and we cannot distinguish cases where stats is not available and where it is zero. And also in `EntryView` public api, we return -1, returning 0 also breaks that api. Seems it is better to keep it as is.

Reply:
@ahmetmircik would you prefer to introduce the `SimpleRecord#getHits()` that returns 0?

Reply:
Returning `UNSET` is fine. By this we can distinguish cases where `perEntryStatsEnabled` is true or false. It can only return 0 when `perEntryStatsEnabled` is true.

Reply:
> By this we can distinguish cases where `perEntryStatsEnabled` is true or false

Is that a bit hackish way of determining that?

Reply:
No, it is not, we create records based on `perEntryStatsEnabled` flag. What i was trying to say is, having an impossible value, it is -1 in this case, makes issues visible in the code as in this issue. Here when `perEntryStatsEnabled` is `false` we don't expect any hit. Otherwise, this issue could still be a latent error.